addons/mail/models/mail_thread.py

```diff
@@ -2249,6 +2249,26 @@ def _message_compute_parent_id(self, parent_id):
                 parent_id = current_ancestor.parent_id.id if current_ancestor.parent_id else parent_id
         return parent_id
 
+    def _message_compute_subject(self, record_names=None, model_description=None):
+        """ Helper to get the default subject when creating a message on this thread
+            For most records it should just be the name of the record,
+            override in child model when this does not make sense
+            :param record_name: {thread_id -> record_name} for records which were already computed
+            :return: {thread_id -> record_name} for every records
+        """
+        record_names = dict(record_names) if record_names else dict()
+
+        to_compute_ids = [thread_id for thread_id in self.ids if not record_names.get(thread_id)]
+        record_name_gets = {key: value for key, value in self.browse(to_compute_ids).name_get() if value}
```

Comment:
You can use a filtered (like ``missing_names = self.fitered(lambda rec: not record_names.get(rec.id)).name_get()``)